logger.tf

```diff
@@ -0,0 +1,49 @@
+locals {
+  api_mgmt_name = var.apim_suffix == "" ? "cft-api-mgmt-${var.environment}" : "cft-api-mgmt-${var.apim_suffix}"
+  api_mgmt_resource_group = "cft-${var.environment}-network-rg"
+  api_mgmt_logger_name = "${local.api_mgmt_name}-logger"
+  api_mgmt_api_name = "${var.product}-${var.component}-api"
+}
+
+# Configure Application insights logging for API
+# Terraform doesn't currently provide an azurerm_api_management_logger data source, so instead of using
+# the value of an output variable for the api_management_logger_id parameter it has to be set explicitly.
+resource "azurerm_api_management_api_diagnostic" "api_mgmt_api_diagnostic" {
+  identifier               = "applicationinsights"
+  api_management_logger_id = "/subscriptions/${var.aks_subscription_id}/resourceGroups/${local.api_mgmt_resource_group}/providers/Microsoft.ApiManagement/service/${local.api_mgmt_name}/loggers/${local.api_mgmt_logger_name}"
+  api_management_name      = local.api_mgmt_name
+  api_name                 = local.api_mgmt_api_name
+  resource_group_name      = local.api_mgmt_resource_group
+
+  sampling_percentage       = 100.0
```

Comment:
This will log all messages.  If you don't want this to be the default, add a variable for it.